### PR TITLE
RPRBLND-1997: Improve Matlib progress + status update

### PR DESCRIPTION
**Pull request by VascoPi**
PURPOSE
Improve status update, progress. Handle connection error.

EFFECT OF CHANGE
Added Material Library subpanel Tools with Reload Library button. Improved status. Update sync and download progress. Handle connection error.

TECHNICAL STEPS
Added function update_ui.
Added handle for connection errors.
Added operator HDUSD_MATERIAL_OP_matlib_load.
Added subpanel HDUSD_MATLIB_PT_matlib_tools

### CODE REVIEW

**Review by bnagirniak — APPROVED** [review]
Added some improvements in statuses text + updated Matlib URL.

File: src/hdusd/utils/matlib.py [hunk 1]
```diff
@@ -343,58 +355,74 @@ def material_load(mat):
...
+                except requests.exceptions.RequestException as err:
+                    executor.shutdown(wait=True, cancel_futures=True)
+                    self.set_status(f"Connection error. Synced {len(self.materials)} materials")
+                    log.error(err)
+
+                finally:
+                    self.is_synced = True
 
         self.load_thread = threading.Thread(target=load)
+        self.load_thread.daemon = True
```
Comment (DagerD) [inline src/hdusd/utils/matlib.py:425]: You can pass daemon through argument

**Review by DagerD — APPROVED** [review]
Tested. Works ok.

File: src/hdusd/utils/__init__.py [hunk 2]
```diff
@@ -130,3 +130,12 @@ def pass_node_reroute(link):
...
     return link if link.is_valid else None
+
+
+def update_ui(area_type='PROPERTIES', region_type='WINDOW'):
+    for window in bpy.context.window_manager.windows:
+        for area in window.screen.areas:
+            if area.type == area_type:
+                for region in area.regions:
+                    if region.type == region_type:
+                        region.tag_redraw()
```
[the remark this answers is not in the document]
Comment (VascoPi) [reply src/hdusd/utils/__init__.py:141]: `break `will update only the first area, it won't work here as expected.

File: src/hdusd/utils/matlib.py [hunk 3]
```diff
@@ -320,10 +329,13 @@ def categories_list(self):
         # required for thread safe purposes
         return list(manager.categories.values())
 
-    def check_load_materials(self):
-        if self.materials is not None:
+    def check_load_materials(self, reset=False):
+        if self.materials is not None and not reset:
```
[the remark this answers is not in the document]
Comment (bnagirniak) [reply src/hdusd/utils/matlib.py:333]: Here we should explicitly check `if self.materials is not None` better add comment here

File: src/hdusd/ui/matlib.py [hunk 4]
```diff
@@ -86,8 +96,10 @@ def execute(self, context):
 
 
 class HDUSD_MATLIB_PT_matlib(HdUSD_Panel):
+    bl_idname = "HDUSD_MATLIB_PT_matlib"
     bl_label = "Material Library"
     bl_context = "material"
+    bl_region_type = 'WINDOW'
```
[the remark this answers is not in the document]
Comment (VascoPi) [reply src/hdusd/ui/matlib.py:102]: bl_region_type is a key, and bl_context is a string type (blender API).

File: src/hdusd/utils/matlib.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (VascoPi) [reply src/hdusd/utils/matlib.py:333]: This made for the first start of the manager to prevent the next update if no materials found.

**Comment by bsavery** [comment]
@VascoPi ready to merge?

**Comment by VascoPi** [comment]
> @VascoPi ready to merge?

Need to check one more issue after URL update.

**Comment by VascoPi** [comment]
> @VascoPi ready to merge?

Yes.

**Pull request merged**